### PR TITLE
Add support for Teal language

### PR DESCRIPTION
**Pull request by barsoosayque**
Add basic support for [Teal language](https://github.com/teal-language/tl).
> The compiler for Teal, a typed dialect of Lua 

Queries and grammar are from [euclidianAce/tree-sitter-teal](https://github.com/euclidianAce/tree-sitter-teal). Language server is [tl/teal-language-server](https://github.com/teal-language/teal-language-server). I couldn't find a proper formatter.

### CODE REVIEW

File: languages.toml
Comment (the-mikedavis) [inline languages.toml:1282]: It looks like the tree-sitter-teal repo doesn't have a LICENSE yet. Could you propose adding one and update the commit here once one is merged?

File: languages.toml
Comment (the-mikedavis) [inline languages.toml:1294]:
```suggestion
scope = "source.tl"
injection-regex = "teal"
file-types = ["tl"]
comment-tokens = "--"
block-comment-tokens = { start = "--[[", end = "--]]" }
roots = [ "tlconfig.lua" ]
language-servers = [ "teal-lsp" ]
```

these are the default values for `auto-format` and `grammar` so they can be omitted

File: runtime/queries/teal/highlights.scm
Comment (the-mikedavis) [inline runtime/queries/teal/highlights.scm:1]: The captures in this file will need to be updated to use our capture names: https://docs.helix-editor.com/master/themes.html#syntax-highlighting

For example `@boolean` should become `@constant.character.boolean`, `@parameter` should become `@variable.parameter`

File: runtime/queries/teal/locals.scm
Comment (the-mikedavis) [inline runtime/queries/teal/locals.scm:1]: For the locals file we follow tree-sitter https://tree-sitter.github.io/tree-sitter/syntax-highlighting#local-variables

So only `@local.definition`, `@local.reference` and `@local.scope` are allowed

File: languages.toml
Comment (barsoosayque) [reply languages.toml:1282]: Issue filed: https://github.com/euclidianAce/tree-sitter-teal/issues/36. Though, I'm not sure how active is the maintainer, I hope it will be resolved soon.

**Comment by barsoosayque** [comment]
I've updated `highlights.scm` and `locals.scm` to the best of my abilities to match the required syntax; never worked with tree-sitter before. Highlighting in helix looks alright for my teal scripts, nothing seems to be broken

File: languages.toml
Comment (the-mikedavis) [reply languages.toml:1282]: Yep that's perfect. Now we can bump the revision to include the license:

```suggestion
source = { git = "https://github.com/euclidianAce/tree-sitter-teal", rev = "3db655924b2ff1c54fdf6371b5425ea6b5dccefe" }
```

File: runtime/queries/teal/highlights.scm
Comment (the-mikedavis) [inline runtime/queries/teal/highlights.scm:171]: In the long run we're considering adding a per-language config option for highlighting syntax errors from tree-sitter. In the meantime we usually remove this pattern since the syntax error highlights can be very noisy

```suggestion
```

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**
